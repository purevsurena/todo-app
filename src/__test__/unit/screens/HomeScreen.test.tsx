import React from "react";
import { render } from "@testing-library/react-native";
import HomeScreen from "@/app/(tabs)/index";

describe("HomeScreen Component", () => {
  it("renders correctly and matches the snapshot", () => {
    const { toJSON } = render(<HomeScreen />);
    expect(toJSON()).toMatchSnapshot();
  });
});
