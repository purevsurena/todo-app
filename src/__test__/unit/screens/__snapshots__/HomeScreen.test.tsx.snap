// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen Component renders correctly and matches the snapshot 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#fff",
      },
      {
        "backgroundColor": "#000",
        "flex": 1,
        "padding": 16,
        "paddingTop": 64,
      },
    ]
  }
>
  <RCTScrollView>
    <View>
      <View
        style={
          [
            {
              "height": 24,
            },
            undefined,
          ]
        }
        testID="white-space"
      />
      <View
        style={
          [
            {
              "display": "flex",
            },
            {
              "alignItems": undefined,
              "alignSelf": undefined,
              "flex": undefined,
              "flexDirection": "row",
              "justifyContent": undefined,
            },
            undefined,
          ]
        }
        testID="flexComponent"
      >
        <Text
          style={
            [
              {
                "color": "#11181C",
              },
              undefined,
              undefined,
              undefined,
              {
                "fontSize": 20,
                "fontWeight": "bold",
              },
              undefined,
              undefined,
            ]
          }
        >
          Hello,
        </Text>
        <View
          collapsable={false}
          jestAnimatedStyle={
            {
              "value": {
                "transform": [
                  {
                    "rotate": "0deg",
                  },
                ],
              },
            }
          }
          style={
            {
              "transform": [
                {
                  "rotate": "0deg",
                },
              ],
            }
          }
        >
          <Text
            style={
              [
                {
                  "color": "#11181C",
                },
                {
                  "fontSize": 16,
                  "lineHeight": 24,
                },
                undefined,
                undefined,
                undefined,
                undefined,
                {
                  "fontSize": 28,
                  "lineHeight": 32,
                  "marginTop": -6,
                },
              ]
            }
          >
            ðŸ‘‹
          </Text>
        </View>
      </View>
      <Text
        style={
          [
            {
              "color": "#11181C",
            },
            {
              "fontSize": 16,
              "lineHeight": 24,
            },
            undefined,
            undefined,
            undefined,
            undefined,
            {
              "color": "#9E9EA5",
              "fontSize": 14,
              "fontWeight": "500",
            },
          ]
        }
      >
        Adventure is waiting...
      </Text>
      <View
        style={
          [
            {
              "height": 24,
            },
            undefined,
          ]
        }
        testID="white-space"
      />
      <View
        style={
          [
            {
              "display": "flex",
            },
            {
              "alignItems": undefined,
              "alignSelf": undefined,
              "flex": undefined,
              "flexDirection": "row",
              "justifyContent": "space-between",
            },
            undefined,
          ]
        }
        testID="flexComponent"
      >
        <View
          style={
            [
              {
                "backgroundColor": "#fff",
              },
              {
                "backgroundColor": "#1c1c1e",
                "borderRadius": 16,
                "padding": 12,
                "paddingTop": 24,
                "paddingVertical": 16,
                "width": 351,
              },
            ]
          }
        >
          <View
            style={
              {
                "flexDirection": "row",
                "justifyContent": "space-between",
                "paddingBottom": 12,
              }
            }
          >
            <Text />
            <Text
              style={
                [
                  {
                    "color": "#11181C",
                  },
                  undefined,
                  {
                    "fontSize": 32,
                    "fontWeight": "bold",
                    "lineHeight": 32,
                  },
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                ]
              }
            >
              3
            </Text>
          </View>
          <Text
            style={
              [
                {
                  "color": "#11181C",
                },
                undefined,
                undefined,
                undefined,
                {
                  "fontSize": 20,
                  "fontWeight": "bold",
                },
                undefined,
                undefined,
              ]
            }
          >
            Today
          </Text>
        </View>
        <View
          style={
            [
              {
                "backgroundColor": "#fff",
              },
              {
                "backgroundColor": "#1c1c1e",
                "borderRadius": 16,
                "padding": 12,
                "paddingTop": 24,
                "paddingVertical": 16,
                "width": 351,
              },
            ]
          }
        >
          <View
            style={
              {
                "flexDirection": "row",
                "justifyContent": "space-between",
                "paddingBottom": 12,
              }
            }
          >
            <Text />
            <Text
              style={
                [
                  {
                    "color": "#11181C",
                  },
                  undefined,
                  {
                    "fontSize": 32,
                    "fontWeight": "bold",
                    "lineHeight": 32,
                  },
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                ]
              }
            >
              0
            </Text>
          </View>
          <Text
            style={
              [
                {
                  "color": "#11181C",
                },
                undefined,
                undefined,
                undefined,
                {
                  "fontSize": 20,
                  "fontWeight": "bold",
                },
                undefined,
                undefined,
              ]
            }
          >
            Tomorrow
          </Text>
        </View>
      </View>
      <View
        style={
          [
            {
              "height": 24,
            },
            undefined,
          ]
        }
        testID="white-space"
      />
      <Text
        style={
          [
            {
              "color": "#11181C",
            },
            undefined,
            undefined,
            undefined,
            {
              "fontSize": 20,
              "fontWeight": "bold",
            },
            undefined,
            undefined,
          ]
        }
      >
        My List
      </Text>
      <View
        style={
          [
            {
              "height": 24,
            },
            undefined,
          ]
        }
        testID="white-space"
      />
      <View
        style={
          [
            {
              "display": "flex",
            },
            {
              "alignItems": undefined,
              "alignSelf": undefined,
              "flex": undefined,
              "flexDirection": "row",
              "justifyContent": "space-between",
            },
            undefined,
          ]
        }
        testID="flexComponent"
      >
        <View
          style={
            [
              {
                "backgroundColor": "#fff",
              },
              {
                "backgroundColor": "#1c1c1e",
                "borderRadius": 16,
                "padding": 12,
                "paddingTop": 24,
                "paddingVertical": 16,
                "width": 351,
              },
            ]
          }
        >
          <View
            style={
              {
                "flexDirection": "row",
                "justifyContent": "space-between",
                "paddingBottom": 12,
              }
            }
          >
            <Text />
            <Text
              style={
                [
                  {
                    "color": "#11181C",
                  },
                  undefined,
                  {
                    "fontSize": 32,
                    "fontWeight": "bold",
                    "lineHeight": 32,
                  },
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                ]
              }
            >
              3
            </Text>
          </View>
          <Text
            style={
              [
                {
                  "color": "#11181C",
                },
                undefined,
                undefined,
                undefined,
                {
                  "fontSize": 20,
                  "fontWeight": "bold",
                },
                undefined,
                undefined,
              ]
            }
          >
            ToDo
          </Text>
        </View>
        <View
          style={
            [
              {
                "backgroundColor": "#fff",
              },
              {
                "backgroundColor": "#1c1c1e",
                "borderRadius": 16,
                "padding": 12,
                "paddingTop": 24,
                "paddingVertical": 16,
                "width": 351,
              },
            ]
          }
        >
          <View
            style={
              {
                "flexDirection": "row",
                "justifyContent": "space-between",
                "paddingBottom": 12,
              }
            }
          >
            <Text />
            <Text
              style={
                [
                  {
                    "color": "#11181C",
                  },
                  undefined,
                  {
                    "fontSize": 32,
                    "fontWeight": "bold",
                    "lineHeight": 32,
                  },
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                ]
              }
            >
              0
            </Text>
          </View>
          <Text
            style={
              [
                {
                  "color": "#11181C",
                },
                undefined,
                undefined,
                undefined,
                {
                  "fontSize": 20,
                  "fontWeight": "bold",
                },
                undefined,
                undefined,
              ]
            }
          >
            Scheduled
          </Text>
        </View>
      </View>
      <View
        style={
          [
            {
              "height": 24,
            },
            undefined,
          ]
        }
        testID="white-space"
      />
      <Text
        style={
          [
            {
              "color": "#11181C",
            },
            undefined,
            undefined,
            undefined,
            {
              "fontSize": 20,
              "fontWeight": "bold",
            },
            undefined,
            undefined,
          ]
        }
      >
        Quck notes
      </Text>
      <View
        style={
          [
            {
              "height": 24,
            },
            undefined,
          ]
        }
        testID="white-space"
      />
      <View
        style={
          [
            {
              "backgroundColor": "#fff",
            },
            {
              "backgroundColor": "#1c1c1e",
              "borderRadius": 16,
              "padding": 12,
              "paddingTop": 24,
              "paddingVertical": 16,
              "width": 718,
            },
          ]
        }
      >
        <View
          style={
            {
              "flexDirection": "row",
              "justifyContent": "space-between",
              "paddingBottom": 12,
            }
          }
        >
          <Text />
          <Text
            style={
              [
                {
                  "color": "#11181C",
                },
                undefined,
                {
                  "fontSize": 32,
                  "fontWeight": "bold",
                  "lineHeight": 32,
                },
                undefined,
                undefined,
                undefined,
                undefined,
              ]
            }
            testID="cardTitle"
          >
            12
          </Text>
        </View>
        <Text
          style={
            [
              {
                "color": "#11181C",
              },
              undefined,
              undefined,
              undefined,
              {
                "fontSize": 20,
                "fontWeight": "bold",
              },
              undefined,
              undefined,
            ]
          }
          testID="cardSubtitle"
        >
          Quick notes write here...
        </Text>
        <View
          style={
            [
              {
                "height": 4,
              },
              undefined,
            ]
          }
          testID="white-space"
        />
      </View>
    </View>
  </RCTScrollView>
  <View>
    <Text />
  </View>
  <Text
    style={
      [
        {
          "color": "#11181C",
        },
        {
          "fontSize": 16,
          "lineHeight": 24,
        },
        undefined,
        undefined,
        undefined,
        undefined,
        {
          "fontSize": 16,
          "fontWeight": "500",
          "marginLeft": 4,
        },
      ]
    }
  >
    New Reminder
  </Text>
</View>
`;
